<a href="https://colab.research.google.com/github/kevmanning/DS-Unit-1-Sprint-1-Data-Wrangling-and-Storytelling/blob/master/DS_111_Exploratory_Data_Analysis_Assignment_Kevin_Manning.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Learning Objectives

# Exploratory Data Analysis

- Objective 01 - submit assignments and save work via GitHub
- Objective 02 - load a CSV dataset from a URL using `pandas.read_csv`
- Objective 03 - load a CSV dataset from a local file using `pandas.read_csv
- Objective 04 - use basic Pandas functions for Exploratory Data Analysis-EDA
- Objective 05 - describe and discriminate between basic data types such as categorical, quantitative, continuous, discrete, ordinal, nominal and identifier



#Explore the University of California - Irvine Adult Dataset

###1) Load a dataset via its URL.

Load the Adult dataset from: 'https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.data'

Specify the column headers as:'age', 'workclass', 'fnlwgt', 'education', 'education-num', 'marital-status', 'occupation', 'relationship', 'race', 'sex', 'capital-gain', 'capital-loss', 'hours-per-week','native-country', 'income']

In [2]:
#Code here
import pandas as pd
adult_df_url = 'https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.data'

adult_df = pd.read_csv(adult_df_url)

###2) Print the first 5 and the last 5 rows.

In [3]:
#Code here
adult_df.head()

39          State-gov   77516   Bachelors  ...   0  40   United-States   <=50K
0  50   Self-emp-not-inc   83311   Bachelors  ...   0  13   United-States   <=50K
1  38            Private  215646     HS-grad  ...   0  40   United-States   <=50K
2  53            Private  234721        11th  ...   0  40   United-States   <=50K
3  28            Private  338409   Bachelors  ...   0  40            Cuba   <=50K
4  37            Private  284582     Masters  ...   0  40   United-States   <=50K

[5 rows x 15 columns]

In [4]:
#Code here
adult_df.tail()

39      State-gov   77516    Bachelors  ...   0  40   United-States   <=50K
32555  27        Private  257302   Assoc-acdm  ...   0  38   United-States   <=50K
32556  40        Private  154374      HS-grad  ...   0  40   United-States    >50K
32557  58        Private  151910      HS-grad  ...   0  40   United-States   <=50K
32558  22        Private  201490      HS-grad  ...   0  20   United-States   <=50K
32559  52   Self-emp-inc  287927      HS-grad  ...   0  40   United-States    >50K

[5 rows x 15 columns]

###3) Determine the data type of each varaible.

In [5]:
#Code here
adult_df.dtypes

39                 int64
 State-gov        object
 77516             int64
 Bachelors        object
 13                int64
 Never-married    object
 Adm-clerical     object
 Not-in-family    object
 White            object
 Male             object
 2174              int64
 0                 int64
 40                int64
 United-States    object
 <=50K            object
dtype: object

###4) Determine the dimensions of the dataset.

In [6]:
adult_df.shape

(32560, 15)

###5) Are there any missing values in the dataset?
Check for missing values using .isnull().sum().

In [7]:
#Code here
adult_df.isnull().sum()


39                0
 State-gov        0
 77516            0
 Bachelors        0
 13               0
 Never-married    0
 Adm-clerical     0
 Not-in-family    0
 White            0
 Male             0
 2174             0
 0                0
 40               0
 United-States    0
 <=50K            0
dtype: int64

###6) Use the .describe() function in order to see the summary statistics of the numeric columns.

Are there any values shown that might be a code for missing data?  We'll learn how to change the values in a DataFrame tomorrow.

In [8]:
#Code here
adult_df.describe()

39         77516  ...             0            40
count  32560.000000  3.256000e+04  ...  32560.000000  32560.000000
mean      38.581634  1.897818e+05  ...     87.306511     40.437469
std       13.640642  1.055498e+05  ...    402.966116     12.347618
min       17.000000  1.228500e+04  ...      0.000000      1.000000
25%       28.000000  1.178315e+05  ...      0.000000     40.000000
50%       37.000000  1.783630e+05  ...      0.000000     40.000000
75%       48.000000  2.370545e+05  ...      0.000000     45.000000
max       90.000000  1.484705e+06  ...   4356.000000     99.000000

[8 rows x 6 columns]

Answer question here.
in the '2174' column, the 99999. maximum value, while from minimum to
the 75% are all 0

###7) Use the .describe(exclude='number') function to calculate summary statistics for the non-numeric columns




In [9]:
#Code here
adult_df.describe(exclude = 'number')

State-gov  Bachelors  ...   United-States   <=50K
count       32560      32560  ...           32560   32560
unique          9         16  ...              42       2
top       Private    HS-grad  ...   United-States   <=50K
freq        22696      10501  ...           29169   24719

[4 rows x 9 columns]

###8) Calculate the number of observations in each category of the relationship variable.

In [10]:
#Code here
adult_df[' Bachelors'].value_counts()

 HS-grad         10501
 Some-college     7291
 Bachelors        5354
 Masters          1723
 Assoc-voc        1382
 11th             1175
 Assoc-acdm       1067
 10th              933
 7th-8th           646
 Prof-school       576
 9th               514
 12th              433
 Doctorate         413
 5th-6th           333
 1st-4th           168
 Preschool          51
Name:  Bachelors, dtype: int64

In [11]:
adult_df[' Never-married'].value_counts()

 Married-civ-spouse       14976
 Never-married            10682
 Divorced                  4443
 Separated                 1025
 Widowed                    993
 Married-spouse-absent      418
 Married-AF-spouse           23
Name:  Never-married, dtype: int64

In [12]:
adult_df[' Not-in-family'].value_counts()

 Husband           13193
 Not-in-family      8304
 Own-child          5068
 Unmarried          3446
 Wife               1568
 Other-relative      981
Name:  Not-in-family, dtype: int64

###9) Select and print the occupation category.

In [13]:
#Code here
print(adult_df[' Adm-clerical'])

0           Exec-managerial
1         Handlers-cleaners
2         Handlers-cleaners
3            Prof-specialty
4           Exec-managerial
                ...        
32555          Tech-support
32556     Machine-op-inspct
32557          Adm-clerical
32558          Adm-clerical
32559       Exec-managerial
Name:  Adm-clerical, Length: 32560, dtype: object


#Load a dataset saved on your personal computer

Go to the website http://users.stat.ufl.edu/~winner/datasets.html and scroll down to **NASCAR Race Results by Driver 1975-2003**

Download and save the .csv file to your personal computer.

###10) Upload the dataset using the following commands.

In [14]:
nascar_df = pd.read_csv('nascard.csv')

In [15]:
#Code here


###11) Print the first 10 rows of the dataset

In [16]:
#Code here
nascar_df.head(10)

serRace  year  yearRace  finish  ...  prize  carsRace    carMake          driver
0        1  1975         1       1  ...  12035        35    Matador    BobbyAllison
1        1  1975         1       2  ...   8135        35    Mercury    DavidPearson
2        1  1975         1       3  ...   6535        35  Chevrolet     CecilGordon
3        1  1975         1       4  ...   5035        35      Dodge      DaveMarcis
4        1  1975         1       5  ...   3835        35       Ford     ElmoLangley
5        1  1975         1       6  ...   2885        35  Chevrolet     JamesHylton
6        1  1975         1       7  ...   3185        35      Dodge    RichardPetty
7        1  1975         1       8  ...   2485        35  Chevrolet    GaryMatthews
8        1  1975         1       9  ...   2335        35      Dodge         EdNegre
9        1  1975         1      10  ...   2185        35  Chevrolet  HershelMcGriff

[10 rows x 10 columns]

###12) Answer the following questions for the variables prize, carMake and driver:


*   Is the variable categorical or quantitative?
*   If the variable is categorical, determine if it is ordinal, nominal or an identifier variable.
*   If the vriable is quantitative, determine if it is continuous or discrite.





In [17]:
nascar_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34884 entries, 0 to 34883
Data columns (total 10 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   serRace   34884 non-null  int64 
 1   year      34884 non-null  int64 
 2   yearRace  34884 non-null  int64 
 3   finish    34884 non-null  int64 
 4   start     34884 non-null  int64 
 5   laps      34884 non-null  int64 
 6   prize     34884 non-null  int64 
 7   carsRace  34884 non-null  int64 
 8   carMake   34884 non-null  object
 9   driver    34884 non-null  object
dtypes: int64(8), object(2)
memory usage: 2.7+ MB


In [18]:
nascar_df['prize'].describe()

count    3.488400e+04
mean     2.704011e+04
std      4.455498e+04
min      0.000000e+00
25%      3.835000e+03
50%      1.201000e+04
75%      3.649625e+04
max      2.277975e+06
Name: prize, dtype: float64

In [19]:
nascar_df['carMake'].describe()

count         34884
unique           10
top       Chevrolet
freq          12663
Name: carMake, dtype: object

In [20]:
nascar_df['driver'].describe()

count         34884
unique          683
top       RickyRudd
freq            803
Name: driver, dtype: object

#Answer here.
Prize: quantitative; continuous, can be more precisely measured

Car Make: categorical; nominal, no natural ordering

Driver: categorical; identifier, identify each record uniquely

#Data Storytelling Project Milestone

Begin looking at Data Storytelling (previously called Build Week) [project info](https://github.com/LambdaSchool/DS-Unit-1-Build) and at projects created by previous Lambda students.

[Projects — Lambda DS1!](https://drive.google.com/file/d/1RJoAqDgawzSHOJPlDQRRlfz-EnIsskpL/view)